src/openforms/registrations/contrib/objects_api/tests/test_backend.py

```diff
@@ -220,3 +226,199 @@ def test_registration_works_after_failure(self, m: requests_mock.Mocker):
+                api_root="http://localhost:8001/api/v2/",
+                api_type=APITypes.orc,
+                oas="https://example.com/",
+                header_key="Authorization",
+                # See the docker compose fixtures:
+                header_value="Token 171be5abaf41e7856b423ad513df1ef8f867ff48",
+                auth_type=AuthTypes.api_key,
+            ),
+            objects_service=ServiceFactory.create(
+                api_root="http://localhost:8002/api/v2/",
+                api_type=APITypes.orc,
+                oas="https://example.com/",
+                header_key="Authorization",
+                # See the docker compose fixtures:
+                header_value="Token 7657474c3d75f56ae0abd0d1bf7994b09964dca9",
+                auth_type=AuthTypes.api_key,
+            ),
+            drc_service=ServiceFactory.create(
+                api_root="http://localhost:8003/documenten/api/v1/",
+                api_type=APITypes.drc,
+                # See the docker compose fixtures:
+                client_id="test_client_id",
+                secret="test_secret_key",
+                auth_type=AuthTypes.zgw,
+            ),
+            catalogi_service=ServiceFactory.create(
+                api_root="http://localhost:8003/catalogi/api/v1/",
+                api_type=APITypes.ztc,
+                # See the docker compose fixtures:
+                client_id="test_client_id",
+                secret="test_secret_key",
+                auth_type=AuthTypes.zgw,
+            ),
+        )
+
+    def test_submission_with_objects_api_backend_create_and_update_object(self):
+        plugin = ObjectsAPIRegistration(PLUGIN_IDENTIFIER)
+
+        with self.subTest("Create an object"):
+            submission_create = SubmissionFactory.from_components(
+                [
+                    {
+                        "key": "voornaam",
+                        "registration": {
+                            "attribute": RegistrationAttribute.initiator_voornamen,
+                        },
+                    },
+                ],
+                submitted_data={"voornaam": "Foo"},
+                form_definition_kwargs={"slug": "fd-create"},
+            )
+            object_create_result = plugin.register_submission(
+                submission_create,
+                {
+                    "version": 1,
+                    "objecttype": UUID("8e46e0a5-b1b4-449b-b9e9-fa3cea655f48"),
+                    "objecttype_version": 3,
+                    "objects_api_group": self.objects_api_group,
+                    "update_existing_object": False,
+                },
+            )
+
+            assert object_create_result
+
+            self.assertEqual(
+                object_create_result["record"]["data"]["data"]["fd-create"]["voornaam"],
+                "Foo",
+            )
+
+        with self.subTest("Update the above existing object"):
+            submission_update = SubmissionFactory.from_components(
```

Comment:
Question: `SubmissionFactory.from_components` will create a new form, different from the first one. Does this reflect the expected flow? From what I read in the feature request, seems to be the case (a different form is used but they want to update the same registration object) but just wanted to be sure I'm not mistaken.

Reply:
I don't think that affects the desired result but indeed a nice test would be one form with multiple submissions I guess. I have updated this.

Reply:
Hum I don't think it had to be changed then, the initial flow you had created different forms, so I just wanted to make sure it reflected the real use case hence my question

Reply:
I discussed with Sergei about this:

In fact it doesn't matter who/what created the initial data record (it could be from another form from OF, but also an external party). So to simplify the test I could just use the objects client to create the initial data, and then test the flow we are interested in (this is done in a similar way [here](https://github.com/open-formulieren/open-forms/blob/6fe0b1ef35709c9772fd1936e2ffb98caf5ba1e3/src/openforms/registrations/contrib/objects_api/tests/test_update_payment_status_v2.py#L43-L69))